knowhere/index/vector_index/IndexHNSW.cpp

```diff
@@ -29,6 +29,21 @@
 #include "index/vector_index/helpers/RangeUtil.h"
 
 namespace knowhere {
+namespace {
+    inline int64_t
+    CheckAndGetEfValue(const Config& config) {
+        auto topk_val = GetMetaTopk(config);
+        if (CheckKeyInConfig(config, indexparam::EF)) {
```

Comment:
suggest move these code to HNSWConfAdapter::CheckSearch
        if (CheckKeyInConfig(config, indexparam::EF)) {
            auto ef_val = GetIndexParamEf(config);
            if (ef_val < topk_val) {
                KNOWHERE_THROW_MSG("ef is smaller than topk in hnsw.");
            }
        }